### DISCUSSION TITLE
Possible to use the tool on an empty repository?

### DISCUSSION THREAD

**Discussion by Tr00d** [murex/TCR#66] [Q&A] [answered]
Hi,

First of all, thanks for building the tool, I'm a big fan and I mostly used the Shell one until recently. I picked up the Go version from version 13.0 with C# support. 
It's now a tool I use in most workshops and katas 👍 

While trying to demo the tool for someone, I realized I was not able to use it on an empty repository. I didn't find anything in the configuration.
Did I miss something, or is it possible at all? 

Repro steps:
- Create new solution
- Execute `git init`
- Run `./tcr.exe -l csharp check`

Here's what I have in the logs:
[TCR] ➤ checking git environment
[TCR]   ✔ git command path is C:\Program Files\Git\cmd\git.exe
[TCR]   ✔ git version is 2.36.0.windows.1
[TCR]   ✔ git username is Guillaume Faas
[TCR]   ▼ **git remote origin is not set**

Thanks a lot and keep up the good work!

**Comment by mengdaming** [comment, marked as answer]
Hi @Tr00d,

First of all, thank you for your interest in TCR tool. We're glad that you like it, and are grateful for the contribution you provided us for C# language.

The issue you're raising is actually a new case for us. Indeed so far we've never faced the need to run TCR on a brand new local repository without any remote already set up.

We believe it's a totally valid case that will bring more flexibility in using the tool, so we'll adjust TCR to support it. The only restriction that will come with it is that all git push/pull/fetch operations will be turned off when no remote is set up.

**Reply by Tr00d** [reply]
Thanks for letting me know. 
It used to work with the Shell version, no? I'm not sure anymore, I'll try it out.

It's a specific purpose for trying the tool, without side effects on a repository. So, of course, git push/pull/fetch has to be disabled when working in "offline" mode. 

Please, let me know when you plan to include that in a version. I'll gladly try it on my side.

**Reply by mengdaming** [reply]
Hi @Tr00d 

We just released [TCR v0.18.0](https://github.com/murex/TCR/releases/tag/v0.18.0) which includes the changes allowing to run TCR on a local repository with no remote configured. Feel free to give it a try, and let us know how it goes.

**Reply by Tr00d** [reply]
Wow, I didn't expect this to come out so soon! Thanks for that 😄 

The process can now start without remote but I do have a little issue though. I ran the following command `.\tcr.exe solo t- dotnet -l csharp`  and the process seems to be ok.
Still, when a change is detected, I have something weird on `dotnet test` (build is fine). I don't understand why it's trying to use `junit` as logger. Is there an additional configuration to bypass that? 

Here's the output:
```
...
[TCR] Running Tests
dotnet test --no-build --logger=junit
C:\xxx (.NETCoreApp,Version=v6.0)
Microsoft (R) Test Execution Command Line Tool Version 17.2.0 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
Could not find a test logger with AssemblyQualifiedName, URI or FriendlyName 'junit'.

[TCR] Some tests are failing! That's unfortunate
...
```

**Reply by mengdaming** [reply]
Do you have JunitXml.TestLogger add-on in your test project? As far as I remember this add-on is required with C# to be able to generate junit-friendly reports. 

You may have a look [here](https://github.com/murex/Kata-BowlingGame/blob/master/csharp/tests/Murex/Kata/BowlingGame/BowlingGame.Tests.csproj) (line 12).

**Reply by mengdaming** [reply]
FYI this junit thing was added by default in recent versions of TCR for capturing test stats in git commit messages. It's still experimental, so a bit rough on the edges and not well documented yet.

If you want to remove the --logger=junit option when running tests, here's what you can do:

Run the following to create configuration files in your project, based on default values:

`.\tcr.exe config save`

All configurations are saved under `.tcr` directory

Edit the file `.tcr/toolchain/dotnet.yml` and remove the `--logger=junit` argument

You can then run TCR as you did before

**Reply by Tr00d** [reply]
Okay, I didn't know about Junit. I may not have been the latest version of the tool...
I tried both ways and everything works like a charm.

Sorry for the hustle and thanks for adding this feature 👏
